// @fixico/meth/Paragraph.js
import React from 'react'
import Radium from 'radium'
import PropTypes from 'prop-types'
import theme from '../../theme'

const styles = {
  default: {
    fontSize: theme.fontSizeM,
    marginBottom: '14px',
    color: theme.black,
    fontFamily: theme.font,
    lineHeight: '1.7em',
  },
  small: {
    fontSize: theme.fontSizeS,
    marginBottom: '12px',
  },
  extraSmall: {
    fontSize: theme.fontSizeXs,
    marginBottom: '10px',
  },
  bold: {
    fontWeight: theme.fontWeightBold,
  },
  greyMedium: {
    color: theme.greyMedium,
  },
  grey: {
    color: theme.greyDark,
  },
  noMargin: {
    marginBottom: '0px',
  },
}

const Paragraph = props => (
  <p
    style={[
      styles.default,
      props.small && styles.small,
      props.extraSmall && styles.extraSmall,
      props.bold && styles.bold,
      props.greyMedium && styles.greyMedium,
      props.grey && styles.grey,
      props.noMargin && styles.noMargin,
      props.style,
    ]}
    className={props.className}
  >
    {props.children}
  </p>
)

Paragraph.defaultProps = {
  className: undefined,
  small: false,
  extraSmall: false,
  bold: false,
  greyMedium: false,
  grey: false,
  noMargin: false,
  style: undefined,
  children: undefined,
}

Paragraph.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  small: PropTypes.bool,
  extraSmall: PropTypes.bool,
  bold: PropTypes.bool,
  greyMedium: PropTypes.bool,
  grey: PropTypes.bool,
  noMargin: PropTypes.bool,
  style: PropTypes.object,
}

export default Radium(Paragraph)
