// @fixico/meth/src/Inputs/TextArea/TextArea.js
import React from 'react'
import PropTypes from 'prop-types'
import { FormControl, FormGroup } from 'react-bootstrap'
import theme from '../../theme'
import FeedbackInput from './../FeedbackInput'

const styles = {
  default: {
    color: theme.black,
    height: '100%',
    backgroundColor: theme.white,
    borderRadius: theme.borderRadius,
    border: `1px solid ${theme.greyDark}`,
    fontFamily: 'OpenSans',
    fontSize: '16px',
    minHeight: '150px',
    resize: 'none',
    padding: '12px',
  },
  success: {
    border: `1px solid ${theme.successMedium}`,
  },
  error: {
    border: `1px solid ${theme.errorMedium}`,
  },
  warning: {
    border: `1px solid ${theme.warningMedium}`,
  },
}

const TextArea = props => (
  <FormGroup controlId="formControlsTextarea" >
    <FormControl
      componentClass="textarea"
      type="text"
      style={{
        ...styles.default,
        ...styles[!props.disabled && props.status],
      }}
      placeholder={props.placeholder}
      disabled={props.disabled}
      {...props.input}
    />
    {props.status && !props.active && !props.disabled
      &&
        <FeedbackInput status={props.status} />
    }
  </FormGroup>
)

TextArea.defaultProps = {
  placeholder: undefined,
  disabled: undefined,
  input: undefined,
  status: undefined,
  active: undefined,
}

TextArea.propTypes = {
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  input: PropTypes.object,
  status: PropTypes.string,
  active: PropTypes.bool,
}

export default TextArea
