// @fixico/meth/src/Buttons/RadioButton/RadioButton.js
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button } from 'react-bootstrap'
import theme from '../../theme'

const style = (props, hover) => {
  const marginBottom = props.noMargin ? '0px' : theme.whitespaceM
  const backgroundColor = hover || props.selected
    ? theme.grey
    : 'transparent'
  const styles = {
    backgroundColor,
    color: theme.black,
    fontFamily: theme.fontHeading,
    border: `1px solid ${theme.greyMedium}`,
    height: '30px',
    width: '100%',
    borderRadius: theme.borderRadius,
    marginBottom,
    fontSize: theme.fontSizeL,
    boxShadow: 'none',
    outline: 'none',
    cursor: 'pointer',
    padding: '0px',
  }
  if (props.disabled) {
    styles.opacity = '0.4'
    styles.pointerEvents = 'none'
  }

  return {
    ...styles,
    ...props.style,
  }
}

const iconStyle = {
  height: '30px',
  marginLeft: '-15px',
}

class RadioButton extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    selected: PropTypes.bool,
    icon: PropTypes.string,
    iconStyle: PropTypes.object,
  }

  static defaultProps = {
    disabled: false,
    selected: false,
    icon: undefined,
    iconStyle: {},
  }

  state = { hover: false }

  hover = (value) => {
    this.setState({ hover: value })
  }

  render() {
    return (
      <Button
        onClick={this.props.disabled ? null : this.props.onClick}
        disabled={this.props.disabled}
        selected={this.props.selected}
        onMouseOver={() => this.hover(true)}
        onMouseOut={() => this.hover(false)}
        onFocus={() => this.hover(true)}
        onBlur={() => this.hover(false)}
        style={style(this.props, this.state.hover)}
      >
        {this.props.icon &&
          <img
            style={{ ...iconStyle, ...this.props.iconStyle }}
            alt="button icon"
            src={this.props.icon}
          />
        }
        {this.props.children}
      </Button>
    )
  }
}

export default RadioButton
