// @fixico/meth/src/Stepper/Stepper.js
import React from 'react'
import PropTypes from 'prop-types'
import theme from '../theme'
import { Icon } from '../index'

const styles = {
  container: {
    width: '100%',
    height: theme.whitespaceS,
    marginBottom: theme.whitespaceM,
    display: 'inline-flex',
  },
  circle: {
    height: theme.whitespaceS,
    width: theme.whitespaceS,
    borderRadius: theme.whitespaceS,
    backgroundColor: theme.infoMedium,
  },
  line: (active) => {
    const backgroundColor = active ? theme.infoMedium : theme.greyMedium
    return {
      height: '5px',
      borderRadius: theme.borderRadius,
      width: '22%',
      margin: '3px',
      backgroundColor,
    }
  },
  flag: (props) => {
    const fill = props.hover ? '#3B9ECF' : '#BFBFBF'
    return {
      width: 'auto',
      height: '16px',
      fill,
    }
  },
}

const Stepper = props => (
  <div style={styles.container}>
    <div style={styles.circle} />
    {props.steps.map((step, i) => {
      let isActive = false
      if (props.active === i + 1) {
        isActive = true
      }
      return (
        <div
          key={step}
          onClick={() => props.onClick(step)}
          onKeyDown={() => props.onClick(step)}
          style={styles.line(isActive)}
        />
      )
    })}
    <div style={{ marginTop: '-3px' }} >
      <Icon viewBox="0 0 48 48" style={styles.flag} hover={props.active === props.steps.length} name="flag_icon" />
    </div>
  </div>
)

Stepper.propTypes = {
  steps: PropTypes.array.isRequired,
  active: PropTypes.number.isRequired,
}


export default Stepper
