// @fixico/meth/src/Buttons/Button/styles.js
import theme from '../../theme'

const styles = {
  default: {
    fontFamily: theme.fontHeading,
    width: '100%',
    marginBottom: theme.whitespaceM,
    letterSpacing: 0,
    textAlign: 'center',
    textDecorator: 'none',
    outline: 'none',
    cursor: 'pointer',
    fontWeight: 500,
    borderRadius: theme.borderRadius,
    display: 'inline-block',
    touchAction: 'manipulation',
    verticalAlign: 'middle',
    whitespace: 'nowrap',
    fontSize: theme.fontSizeM,
    lineHeight: '18px',
    ':disabled': { pointerEvents: 'none', opacity: 0.3, boxShadow: 'none' },
  },
  small: { height: 30 },
  medium: { height: 42 },
  large: { height: 50, fontSize: 19, lineHeight: '23px' },
  primary: {
    backgroundColor: theme.green,
    color: theme.white,
    border: 'none',
    boxShadow: theme.boxShadow,
    ':hover': { backgroundColor: theme.greenDark, boxShadow: 'none' },
    ':active': { backgroundColor: theme.greenDark, boxShadow: 'none' },
    ':focus': { backgroundColor: theme.greyLight, boxShadow: 'none', color: theme.green },
  },
  secondary: {
    backgroundColor: theme.white,
    color: theme.black,
    border: `1px solid ${theme.greyMedium}`,
    boxShadow: 'none',
    ':hover': { backgroundColor: theme.grey, border: `1px solid ${theme.grey}` },
    ':active': { backgroundColor: theme.greenDark, border: `1px solid ${theme.greenDark}` },
    ':focus': { backgroundColor: theme.grey, border: `1px solid ${theme.grey}` },
  },
  outlinePrimary: {
    backgroundColor: theme.white,
    color: theme.green,
    border: `1px solid ${theme.green}`,
    boxShadow: 'none',
    ':hover': {
      backgroundColor: theme.greenDark, boxShadow: 'none', color: theme.white, border: `1px solid ${theme.greenDark}`,
    },
    ':active': {
      backgroundColor: theme.greenDark, boxShadow: 'none', color: theme.white, border: `1px solid ${theme.greenDark}`,
    },
    ':focus': {
      backgroundColor: theme.greyLight, boxShadow: 'none', color: theme.green, border: `1px solid ${theme.greyLight}`,
    },
  },
  cta: {
    backgroundColor: theme.yellow,
    color: theme.black,
    border: 'none',
    boxShadow: theme.boxShadow,
    ':hover': { backgroundColor: theme.yellowDark, boxShadow: 'none' },
    ':active': { backgroundColor: theme.yellowDark, boxShadow: 'none' },
    ':focus': { backgroundColor: theme.greyLight, boxShadow: 'none', color: theme.yellowDark },
  },
  outlineCta: {
    backgroundColor: theme.white,
    color: theme.yellow,
    border: `1px solid ${theme.yellow}`,
    boxShadow: 'none',
    ':hover': { backgroundColor: theme.yellowDark, border: `1px solid ${theme.yellowDark}`, color: theme.black },
    ':active': { backgroundColor: theme.yellowDark, border: `1px solid ${theme.yellowDark}`, color: theme.black },
    ':focus': { backgroundColor: theme.greyLight, border: `1px solid ${theme.greyLight}`, color: theme.yellowDark },
  },
  interface: {
    backgroundColor: 'transparent',
    color: theme.green,
    border: `1px solid ${theme.grey}`,
    boxShadow: 'none',
    ':hover': { backgroundColor: theme.greenDark, border: `1px solid ${theme.greenDark}`, color: theme.white },
    ':active': { backgroundColor: theme.greenDark, border: `1px solid ${theme.greenDark}`, color: theme.white },
    ':focus': { backgroundColor: theme.greenDark, border: `1px solid ${theme.greenDark}`, color: theme.white },
  },
  noMargin: (noMargin) => {
    const marginBottom = noMargin ? '0px' : theme.whitespaceM
    return { marginBottom }
  },
}

export default styles
