// @fixico/meth/src/Tags/Tag/Tag.js
import React from 'react'
import PropTypes from 'prop-types'
import { Icon, Header } from '../../index'
import theme from '../../theme'

const styles = {
  default: color => ({
    position: 'absolute',
    background: 'transparent',
    border: `1px solid ${color}`,
    height: '20px',
    textAlign: 'center',
    borderRadius: '2px',
    padding: '0px 6px',
  }),
}

const icon = props => ({
  fill: props.hover,
  float: 'left',
  width: '15px',
  height: '15px',
  marginTop: '1px',
  marginRight: '3px',
})

const Tag = (props) => {
  const color = props.fontColor || theme.successMedium
  const active = props.fontActive || theme.white
  return (
    <div style={{ ...styles.default(props.active ? active : color), ...props.style }}>
      {props.icon &&
        <Icon viewBox="0 0 48 48" style={icon} hover={props.active ? active : color} name="star_icon" />
      }
      <Header tag="H6" noMargin style={{ color: props.active ? active : color, float: 'left' }} >
        {props.children}
      </Header>
    </div>
  )
}

Tag.defaultProps = {
  active: false,
  fontActive: theme.white,
  fontColor: theme.successMedium,
  style: undefined,
  icon: undefined,
}

Tag.propTypes = {
  active: PropTypes.bool,
  fontActive: PropTypes.string,
  fontColor: PropTypes.string,
  style: PropTypes.object,
  children: PropTypes.node.isRequired,
  icon: PropTypes.bool,
}

export default Tag
