// @fixico/meth/src/Typography/Header/Header.js
import React from 'react'
import PropTypes from 'prop-types'
import theme from './../../theme'

const style = {
  base: props => ({
    marginBottom: props.noMargin ? '0px' : '0.7e.m',
    fontFamily: theme.fontHeading,
    lineHeight: '1.3em',
    marginTop: '0px',
    fontWeight: theme.fontWeightBold,
    color: theme.black,
    letterSpacing: '0',
    ...props.style,
  }),
  H1: { fontSize: theme.fontSizeXxxl, lineHeight: '58px' },
  H2: { fontSize: theme.fontSizeXxl, lineHeight: theme.fontSizeXxl },
  H3: { fontSize: theme.fontSizeXl, lineHeight: '32px' },
  H4: { fontSize: theme.fontSizeL, lineHeight: theme.whitespaceM },
  H5: { fontSize: theme.fontSizeM, lineHeight: '20px' },
  H6: { fontSize: theme.fontSizeS, lineHeight: '18px' },
}

const Header = props => (
  <p style={{ ...style.base(props), ...style[props.tag] }} >
    {props.children}
  </p>
)

Header.defaultProps = { children: undefined }

Header.propTypes = {
  children: PropTypes.any,
  tag: PropTypes.string.isRequired,
}

export default Header
