// @fixico/meth/src/Tags/RatingTag/RatingTag.js
import React from 'react'
import PropTypes from 'prop-types'
import { Icon, Header } from '../../index'
import theme from '../../theme'

const tagStyle = {
  backgroundColor: theme.successMedium,
  position: 'absolute',
  left: theme.whitespaceS,
  bottom: theme.whitespaceS,
  height: '23px',
  borderRadius: '4px',
  width: 'fit-content',
  padding: `0px ${theme.whitespaceXs}`,
}

const font = {
  float: 'left',
  color: theme.white,
}

const icon = () => ({
  fill: theme.white,
  float: 'left',
  width: '15px',
  height: '15px',
  marginTop: '4px',
})

const RatingTag = props => (
  <div style={tagStyle}>
    <Icon viewBox="0 0 50 50" style={icon} name="star_icon" />
    <Header tag="H6" style={{ ...font, margin: '3px 6px 3px 3px' }} noMargin>
      {`${props.rating}`}
    </Header>
    <Header tag="H6" style={{ ...font, paddingTop: '3px' }} noMargin>
      {props.text}
    </Header>
  </div>
)

RatingTag.propTypes = {
  rating: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  text: PropTypes.string.isRequired,
}

export default RatingTag
