// @fixico/meth/src/Buttons/TransparentButton/TransparentButton.js
import React from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'
import { Icon, Header } from '../../index'
import theme from '../../theme'

const iconStyle = () => ({
  marginLeft: theme.whitespaceS,
  height: '100%',
  fill: theme.white,
})

const styles = props => ({
  backgroundColor: 'rgba(0,0,0, 0.7)',
  fontWeight: 400,
  border: 'none',
  height: '40px',
  boxShadow: 'none',
  borderRadius: '0px',
  outline: 'none',
  display: 'inline-block',
  touchAction: 'manipulation',
  verticalAlign: 'middle',
  whitespace: 'nowrap',
  paddingLeft: theme.whitespaceS,
  paddingRight: '0px',
  ':disabled': { pointerEvents: 'none', opacity: 0.3 },
  ':hover': { backgroundColor: theme.greenDark },
  ...props.style,
})

const TransparentButton = props => (
  <button
    style={styles(props)}
    onClick={props.disabled ? null : props.onClick}
    disabled={props.disabled}
  >
    <Header
      noMargin
      tag="H5"
      style={{
        position: 'relative',
        color: theme.white,
        float: 'left',
        top: '8px',
        fontWeight: '500',
      }}
    >
      {props.children}
    </Header>
    <Icon name="arrow_open" style={iconStyle} />
  </button>
)

TransparentButton.defaultProps = { disabled: false }

TransparentButton.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
}

export default Radium(TransparentButton)
