// @fixico/meth/src/Inputs/Input/Input.js
import React from 'react'
import PropTypes from 'prop-types'
import { FormControl, FormGroup } from 'react-bootstrap'
import FeedbackInput from './../FeedbackInput'
import theme from './../../theme'

const styles = {
  default: {
    color: theme.black,
    height: '100%',
    padding: '12px',
    backgroundColor: theme.white,
    borderRadius: theme.borderRadius,
    border: `1px solid ${theme.greyDark}`,
    fontFamily: 'OpenSans',
    fontSize: '16px',
  },
  success: {
    border: `1px solid ${theme.successMedium}`,
  },
  error: {
    border: `1px solid ${theme.errorMedium}`,
  },
  warning: {
    border: `1px solid ${theme.warningMedium}`,
  },
}

const Input = props => (
  <FormGroup>
    <FormControl
      type={props.type}
      style={{
        ...styles.default,
        ...styles[!props.disabled && props.status],
      }}
      placeholder={props.placeholder}
      autoFocus={props.autoFocus}
      maxLength={props.maxLength}
      disabled={props.disabled}
      {...props.input}
    />
    {(props.status && !props.active && !props.disabled)
      &&
        <FeedbackInput status={props.status} />
      }
  </FormGroup>
)

Input.defaultProps = {
  type: 'text',
  placeholder: undefined,
  autoFocus: undefined,
  maxLength: undefined,
  disabled: undefined,
  input: undefined,
  active: undefined,
  status: undefined,
}

Input.propTypes = {
  type: PropTypes.string,
  placeholder: PropTypes.string,
  autoFocus: PropTypes.bool,
  maxLength: PropTypes.number,
  disabled: PropTypes.bool,
  input: PropTypes.object,
  active: PropTypes.bool,
  status: PropTypes.string,
}

export default Input
