// @fixico/meth/src/Typography/Paragraph/index.js
import React from 'react'
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import theme from '../../theme'

const P = styled.p`
    margin-bottom: '14px';
    font-family: ${theme.font};
    line-height: '1.7em';
    color: ${theme.black};
    font-size: ${theme.fontSizeM};
    ${props => props.noMargin && css` margin-bottom: 0; `}
    ${props => props.bold && css` font-weight: ${theme.fontWeightBold}; `}
    ${props => props.grey && css` color: ${theme.greyDark}; `}
    ${props => props.greyMedium && css` color: ${theme.greyMedium}; `}
    ${props => props.small && css` font-size: ${theme.fontSizeS}; `}
    ${props => props.extraSmall && css` font-size: ${theme.fontSizeXs}; `}
  `

const Paragraph = props => (
  <P {...props}>
    {props.children}
  </P>
)

Paragraph.defaultProps = {
  className: undefined,
  small: false,
  extraSmall: false,
  bold: false,
  greyMedium: false,
  grey: false,
  noMargin: false,
  style: undefined,
  children: undefined,
}

Paragraph.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  small: PropTypes.bool,
  extraSmall: PropTypes.bool,
  bold: PropTypes.bool,
  greyMedium: PropTypes.bool,
  grey: PropTypes.bool,
  noMargin: PropTypes.bool,
  style: PropTypes.object,
}

export default Paragraph
