// @fixico/meth/src/Popovers/styles.js
import theme from '../theme'

export const arrow = {
  default: {
    position: 'absolute',
    display: 'block',
    width: 0,
    height: 0,
    borderColor: 'transparent',
    borderStyle: 'solid',
    borderWidth: '11px',
  },
  after: (props) => {
    const after = {
      position: 'absolute',
      display: 'block',
      width: 0,
      height: 0,
      borderStyle: 'solid',
      borderWidth: '12px',
      zIndex: -1,
    }
    switch (props.placement) {
      case 'right':
        return {
          top: '-12px',
          left: '-13.5px',
          borderColor: 'transparent',
          borderRightColor: theme[`${props.tag}`],
          ...after,
        }
      case 'left':
        return {
          top: '-12px',
          right: '-13.5px',
          borderColor: 'transparent',
          borderLeftColor: theme[`${props.tag}`],
          ...after,
        }
      case 'top':
        return {
          bottom: '-13.5px',
          left: '-12px',
          borderColor: 'transparent',
          borderTopColor: theme[`${props.tag}`],
          ...after,
        }
      case 'bottom':
        return {
          top: '-13.5px',
          left: '-12px',
          borderColor: 'transparent',
          borderBottomColor: theme[`${props.tag}`],
          ...after,
        }
      default:
        return ''
    }
  },
  place: (props) => {
    switch (props.placement) {
      case 'right':
        return {
          top: '50%',
          left: '-10px',
          marginTop: '-11px',
          borderRightColor: theme[`${props.tag}Light`],
          borderLeftWidth: '0',
        }
      case 'left':
        return {
          top: '50%',
          right: '-10px',
          marginTop: '-11px',
          borderLeftColor: theme[`${props.tag}Light`],
          borderRightWidth: '0',
        }
      case 'top':
        return {
          left: '50%',
          bottom: '-10px',
          marginLeft: '-11px',
          borderTopColor: theme[`${props.tag}Light`],
          borderBottomWidth: '0',
        }
      case 'bottom':
        return {
          left: '50%',
          top: '0px',
          marginTop: '-11px',
          borderBottomColor: theme[`${props.tag}Light`],
          borderTopWidth: '0',
        }
      default:
        return ''
    }
  },
}

export const popoverStyle = props => ({
  left: '2px',
  border: `1px solid ${theme[`${props.tag}`]}`,
  borderRadius: theme.borderRadius,
  top: `-${theme.whitespaceM}`,
  backgroundColor: theme[`${props.tag}Light`],
  color: theme[`${props.tag}Medium`],
  boxShadow: 'none',
  width: '100%',
  position: 'absolute',
  padding: theme.whitespaceS,
  maxWidth: '316px',
  '@media(max-width: 990px)': { maxWidth: '290px' },
})

export const icon = {
  marginRight: theme.whitespaceXs,
  width: '18px',
  marginTop: '5px',
  float: 'left',
}

export const heading = props => ({
  paddingBottom: '0px',
  color: theme[`${props.tag}Medium`],
  display: '-webkit-box',
  ...props.headingStyle,
})
