// @fixico/meth/src/Titles/PointingTitle/PointingTitle.js
import React from 'react'
import PropTypes from 'prop-types'
import { Header } from '../../index'
import theme from '../../theme'

const triangleStyle = color => ({
  position: 'absolute',
  display: 'block',
  width: 0,
  height: 0,
  borderColor: 'transparent',
  borderStyle: 'solid',
  borderWidth: '11px',
  left: '50%',
  bottom: '-10px',
  marginLeft: '-11px',
  borderTopColor: color,
  borderBottomWidth: '0',
})

const wrapperStyle = color => ({
  position: 'relative',
  backgroundColor: color,
  borderRadius: theme.borderRadius,
  width: '100%',
  padding: theme.whitespaceS,
  marginBottom: theme.whitespaceXs,
  height: '46px',
  textAlign: 'center',
})

const PointingTitle = props => (
  <div style={wrapperStyle(props.color)} >
    <Header tag="H5" noMargin style={{ color: theme.white }} >
      {props.children}
    </Header>
    <div style={triangleStyle(props.color)} />
  </div>
)

PointingTitle.propTypes = { children: PropTypes.any.isRequired, color: PropTypes.string.isRequired }

export default PointingTitle
