// @fixico/meth/src/Inputs/SelectInput/SelectInput.js
import React from 'react'
import Radium from 'radium'
import PropTypes from 'prop-types'
import { FormControl, FormGroup } from 'react-bootstrap'
import theme from './../../theme'

const styles = {
  default: {
    color: theme.black,
    height: '100%',
    padding: '12px',
    backgroundColor: theme.white,
    borderRadius: theme.borderRadius,
    border: `1px solid ${theme.greyDark}`,
    fontFamily: 'OpenSans',
    fontSize: '16px',
  },
  success: {
    border: `1px solid ${theme.successMedium}`,
  },
  error: {
    border: `1px solid ${theme.errorMedium}`,
  },
  warning: {
    border: `1px solid ${theme.warningMedium}`,
  },
}

const SelectInput = props => (
  <FormGroup style={{ marginBottom: '12px' }} controlId="formControlsSelect">
    <FormControl
      disabled={props.disabled}
      type="text"
      componentClass="select"
      style={{
        ...styles.default,
        ...styles[!props.disabled && props.status],
      }}
      placeholder={props.placeholder}
      {...props.input}
    >
      <option value="">{props.placeholder}</option>
      {props.options.map((option, index) => (
        <option key={index} // eslint-disable-line
          value={option.value}
        >
          {option.value}
        </option>
      ))}
    </FormControl>
  </FormGroup>
)

SelectInput.defaultProps = {
  type: 'text',
  placeholder: undefined,
  disabled: undefined,
  input: undefined,
  active: undefined,
  status: undefined,
}

SelectInput.propTypes = {
  type: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  })).isRequired,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  input: PropTypes.object,
  active: PropTypes.bool,
  status: PropTypes.string,
}

export default Radium(SelectInput)
